File: frontend/src/types/DemoRequest.ts
export type DemoModuleCode = string;
export type RequestStatus = "PENDING" | "ACCEPTED" | "REJECTED";
export type EmailStatus = "SENT" | "FAILED" | "DISABLED" | string;

export interface DemoRequest {
  id: number;
  requestCode: string;
  schoolName: string;
  contactPerson: string;
  designation: string;
  email: string;
  phone: string;
  address: string;
  studentCount?: number | null;
  interestedModules: string;
  status: RequestStatus;
  logoUrl?: string | null;
  provisionedSchoolId?: number | null;
  demoSchoolId?: number | null;
  emailStatus?: EmailStatus | null;
  emailError?: string | null;
  acceptedAt?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CreateDemoRequestPayload {
  schoolName: string;
  contactPerson: string;
  designation: string;
  email: string;
  phone: string;
  address: string;
  studentCount?: number;
  interestedModules: DemoModuleCode[];
}

export interface PublicDemoRequestResponse {
  id: number;
  requestCode: string;
  schoolName: string;
  status: string;
  createdAt: string;
}

export interface AcceptDemoResponse {
  accountCreated: boolean;
  schoolId: number;
  demoSchoolId: number;
  schoolCode: string;
  schoolName: string;
  email: string;
  phone: string;
  logoUrl?: string | null;
  username: string;
  password?: string | null;
  startsOn: string;
  expiresOn: string;
  remainingDays: number;
  modules: DemoModuleCode[];
  status: string;
  emailStatus: EmailStatus;
  message: string;
}

export interface UpdateDemoRequestPayload {
  status?: RequestStatus;
  message?: string;
}

File: backend/src/main/java/com/pathshala/entity/DemoRequest.java
package com.pathshala.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "demo_requests")
public class DemoRequest extends BaseEntity {
    @Version
    private Long version;

    @Column(nullable = false, unique = true)
    private String requestCode;

    @Column(nullable = false)
    private String schoolName;

    @Column(nullable = false)
    private String contactPerson;
    @Column(nullable = false)
    private String designation;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String phone;

    private String address;

    @Column(name = "student_count")
    private Integer studentCount;

    @Column(length = 1000)
    private String interestedModules;

    @Column(length = 2000)
    private String message;


    @Column(nullable = false, length = 20)
    private String status = "PENDING";

    @Column(name = "logo_key", length = 500)
    private String logoKey;

    @Column(name = "provisioned_school_id", unique = true)
    private Long provisionedSchoolId;

    @Column(name = "demo_school_id", unique = true)
    private Long demoSchoolId;

    @Column(name = "accepted_at")
    private java.time.Instant acceptedAt;

    @Column(name = "accepted_by")
    private Long acceptedBy;

    @Column(name = "email_status", nullable = false, length = 20)
    private String emailStatus = "PENDING";

    @Column(name = "email_error", length = 500)
    private String emailError;

    @Column(name = "credentials_sent_at")
    private java.time.Instant credentialsSentAt;
}
